### DISCUSSION TITLE
Validation issues using native-image

### DISCUSSION THREAD

**Discussion by ElBlokko** [quarkusio/quarkus#24372] [Q&A] [answered]
Hello, 

I'm trying to build a native-image out of my simple crud application. It contains some jakarta validation and for some reason in a runtime it throws an error that doesn't appear in non native version:

`ERROR [io.qua.ver.htt.run.QuarkusErrorHandler] (executor-thread-0) HTTP Request to /data-set failed, error id: cf25f152-6734-44b8-a3d8-f97acb2dc7e0-1: javax.validation.ValidationException: HV000083: The specified annotation javax.validation.constraints.NotNull defines no attribute 'groups'.`

I can fix that by adding **all annotations** to @RegisterForReflection annotation, but I hope it's not proper solution to fix it since it's imported by quarkus-resteasy. I'm using:

quarkus-resteasy 2.7.4.Final 
graalvm GraalVM CE 22.0.0.2

**Comment by gsmet** [comment, marked as answer]
Are you using the `quarkus-hibernate-validator` extension?

If so, please provide a small reproducer so that I can have a look.

**Reply by ElBlokko** [reply]
yes,

**Reply by ElBlokko** [reply]
https://github.com/ElBlokko/GreetingResource.git

**Reply by gsmet** [reply]
The problem is there: `private static final ValidatorFactory validatorFactory = Validation.buildDefaultValidatorFactory();`

Don't create your own ValidatorFactory, use the one initialized by Quarkus with:
```
@Inject
ValidatorFactory validatorFactory
```

You can even inject a `Validator` if it's easier for you.

**Reply by ElBlokko** [reply]
yes, thanks for support :)
